modules/system/assets/ui/js/select.js

```diff
@@ -87,23 +87,25 @@
                         return $request
                     },
                     processResults: function (data, params) {
-                        var results = data.result;
-                        var options = [];
-
-                        for (var i in results) {
-                            if (results.hasOwnProperty(i)) {
-                                var isObject = i != null && i.constructor.name === 'Object';
```

Comment:
@jimcottrell We'll need to maintain the Object handling, in case some people are working with that format.

Reply:
I'm not clear what you're referring to.

Reply:
Are you referring to the Select2 format as the "Object handling"? That format is just passed through to Select2.

Reply:
Sorry @jimcottrell, I see what you are doing now. Don't worry about my comment.